ood-portal-generator/lib/ood_portal_generator/application.rb

```diff
@@ -29,41 +33,214 @@ def context
         config.file? ? YAML.load_file(config) : {}
       end
 
+      def rpm
+        @rpm.nil? ? false : @rpm
+      end
+
+      def force
+        @force.nil? ? false : @force
+      end
+
+      def detailed_exitcodes
+        @detailed_exitcodes.nil? ? false : @detailed_exitcodes
+      end
+
+      def change_exit
+        detailed_exitcodes ? 3 : 0
+      end
+
+      def skip_exit
+        detailed_exitcodes ? 4 : 0
+      end
+
+      def apache
+        ENV['APACHE'] || OodPortalGenerator.scl_apache ? '/opt/rh/httpd24/root/etc/httpd/conf.d/ood-portal.conf' : '/etc/httpd/conf.d/ood-portal.conf'
+      end
+
+      def apache_bak
+        "#{apache}.#{Time.now.strftime('%Y%m%dT%H%M%S')}"
+      end
+
+      def apache_services
+        if OodPortalGenerator.scl_apache
+          ['httpd24-httpd', 'httpd24-htcacheclean']
+        else
+          ['httpd', 'htcacheclean']
+        end
+      end
+
+      def sum
+        ENV['SUM'] || '/etc/ood/config/ood_portal.sha256sum'
+      end
+
+      def parse_input(input)
+        lines = File.readlines(input)
+        new_lines = []
+        lines.each do |line|
+          next if line =~ /^\s*#/
+          new_lines << line
+        end
+        str = new_lines.join('')
+        str
+      end
```

Comment:
```ruby
# return string contents of file without comments
def parse_input(input)
  File.readlines(f).reject {|line| line =~ /^\s*#/ }.join('')
end
```

Maybe a better name is: `read_file_omitting_comments` since `File.read` returns a string.

Reply:
Changes made.